### PR TITLE
- Use logSafe from utils package

### LINKED ISSUE #1091: Residual calculation
In PK-Sim/MoBi, log residuals are calculated via the defined there `LogSafe `function:
https://github.com/Open-Systems-Pharmacology/OSPSuite.Core/blob/fcd42c611764e4af44f266285175ef66f706fdde/src/OSPSuite.Core/Domain/Services/ParameterIdentifications/ResidualCalculator.cs#L118-L124

with
https://github.com/Open-Systems-Pharmacology/OSPSuite.Core/blob/fcd42c611764e4af44f266285175ef66f706fdde/src/OSPSuite.Core/Domain/Constants.cs#L92

The log residual itself is then calculated as `LogSafe(simulatedValue) - LogSafe(observedValue)`:
https://github.com/Open-Systems-Pharmacology/OSPSuite.Core/blob/fcd42c611764e4af44f266285175ef66f706fdde/src/OSPSuite.Core/Domain/Services/ParameterIdentifications/ResidualCalculatorForOnlyObservedData.cs#L12-L20

This solves different problems, e.g. 
* if simulated values are slightly negative (numerical zero) residual would return NaN otherwise
* if simulated values are positive but extremely small (numerical zero as well), residual might become extremely low (negative) value, causing algorithmic problems etc.

So I would suggest implementing residual calculation in R in the same way?

_Originally posted by @Yuri05 in https://github.com/Open-Systems-Pharmacology/OSPSuite-R/pull/1090#discussion_r947213016_

### PR DESCRIPTION
**Pull request by PavelBal**
- Calculate residuals using ln instead of log10
Fixes #1091, #1087, #1212

### CODE REVIEW

**Comment by PavelBal** [comment]
Natural logarithm is what expected by most people when calculating the residuals, so I guess it is safe to make this switch here.

File: tests/testthat/test-utilities-data-combined.R [hunk 1]
```diff
@@ -83,9 +83,16 @@ test_that(
...
+  "calculateResiduals returns a correct vector of log residuals on example data",{
+  pairedData <- calculateResiduals(myDC, scaling = "log")
+  expectedResiduals <- sapply(seq_along(pairedData$yValuesObserved), function(idx){
+    ospsuite.utils::logSafe(pairedData$yValuesSimulated[[idx]]) -
+    ospsuite.utils::logSafe(pairedData$yValuesObserved[[idx]])
+    })
+
   expect_equal(calculateResiduals(myDC, scaling = "log")$residualValues,
-    c(0, 0.022276400, -0.007178578, 0.00000000, -0.010723855, -20),
+               expectedResiduals,
```
Comment (svavil) [inline tests/testthat/test-utilities-data-combined.R:94]: Do we need to define `expectedResiduals` before using? Might have been left undefined.

Comment (PavelBal) [reply tests/testthat/test-utilities-data-combined.R:94]: Line 88 is defining the expected residuals, or what do you mean?

**Review by Felixmil — APPROVED** [review]

**Comment by PavelBal** [comment]
@Yuri05 Would like to have your "go" on this before merging.

**Review by Yuri05 — APPROVED** [review]

**Pull request merged**